app/services/partners/section_error_service.rb

```diff
@@ -0,0 +1,53 @@
+module Partners
+  # SectionErrorService identifies which sections of the Partner Profile step-wise form
+  # should expand when validation errors occur. This helps users easily locate and fix
+  # fields with errors in specific sections.
+  #
+  # Why this service is needed:
+  # In the Partner Profile step-wise form, fields are grouped within collapsible sections.
+  # When validation errors happen, this service maps the fields with errors to their sections,
+  # ensuring the relevant sections open automatically so users can address the issues directly.
+  #
+  # Usage:
+  #   error_keys = [:website, :pick_up_name, :enable_quantity_based_requests]
+  #   service = Partners::SectionErrorService.new(error_keys)
+  #   sections_with_errors = service.call
+  #   # => ["media_information", "pick_up_person", "partner_settings"]
+  #
+  class SectionErrorService
```

Comment:
Is this basically an overgrown hash lookup? I feel like this can be squished into a single class-level method.

Reply:
Updated with class method.